### PR TITLE
Enable 2023.0.0

### PR DESCRIPTION
**Pull request by artanokhov**
Checklist
* [x] Enable GPU Plugin build
* [x] Use flatbuffers from conda-forge instead pointing to source
* [x] Enable linux aarch64 platform
* [x] Switch to 2023.0.0 OV source when it will be available(can be started for CF)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -180,6 +185,7 @@ outputs:
         - {{ compiler('c') }}
         - cmake >=3.13
       host:
+        - libhwloc >=2.5 # starting from this version hwloc supports hybrid cores
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:188]: we need to remove it, `libhwloc` is now dependency of `tbb`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,47 +1,41 @@
...
-    sha256: b8dc1880d9ab71bd24aa5b2565724c12fb78172613e23ce312e0b98a6d8a0dd7
+  - url: https://github.com/openvinotoolkit/openvino/archive/3ec386a74190ca967da82467de6f212f7c8181c7.tar.gz
+    sha256: 0293be981d3217f6cfa112a4821285121efd84f27910262e95426d162ffa33f6
     folder: openvino_sources
     patches:
-    - patches/001-gflags.patch
-  - url: https://github.com/openvinotoolkit/openvino_contrib/archive/13025cfb0a98a5eb0ed531ea922dcb567b879902.tar.gz  # [aarch64 or arm64]
-    sha256: a6bd13848443a2e8f15ea8c24fb049935753157d233913a2174e5cf080c98c90  # [aarch64 or arm64]
+    - patches/0001-fix-tbb-win.patch
+  - url: https://github.com/openvinotoolkit/openvino_contrib/archive/5556f70f2208849530cfdd7c1b93ea5ac89c2018.tar.gz  # [aarch64 or arm64]
```
[the remark this answers is not in the document]
Comment (artanokhov) [reply recipe/meta.yaml:15]: Yeah, I'll do it when switch to newest OV sources, now I iterative adding feature-by-feature

File: recipe/patches/0001-fix-tbb-win.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/cmake/ie_parallel.cmake b/src/cmake/ie_parallel.cmake
+index 1e6b942f88..30f3ec8ec7 100644
+--- a/src/cmake/ie_parallel.cmake
++++ b/src/cmake/ie_parallel.cmake
+@@ -142,7 +142,7 @@ macro(ov_find_package_tbb)
+                 endif()
+             endforeach()
+ 
+-            if(WIN32 AND TARGET TBB::tbbbind_2_5)
++            if(WIN32 AND TARGET TBB::tbbbind_2_5 AND NOT CPACK_GENERATOR STREQUAL "CONDA-FORGE")
```
[the remark this answers is not in the document]
Comment (artanokhov) [reply recipe/patches/0001-fix-tbb-win.patch:10]: Current state pointing to https://github.com/openvinotoolkit/openvino/commit/3ec386a74190ca967da82467de6f212f7c8181c7
commited on 9 March, I'll remove patch when switch to newest version

**Comment by artanokhov** [comment]
@carterbox @xhochy
I'm just wondering - can we enable ONNX FE with build it from sources since blocker PR https://github.com/conda-forge/onnx-feedstock/pull/85 not merged yet? 
We can switch to the conda-forge ONNX module as soon as necessary cpp-lib will be available.

**Comment by xhochy** [comment]
Please address the one comment I have in the PR and we could proceed there. No need to vendor here.

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin please rerender

**Comment by artanokhov** [comment]
> Please address the one comment I have in the PR and we could proceed there. No need to vendor here.

Yeah, so we noticed 2 things
* This ONNX cpp lib will be available for Linux only
* The ONNX in conda-forge using dynamic protobuf - but we use static one, it can potentially lead to issues on the user side
* We are talking about additional 100-150 KB in case if OpenVINO will proceed with ONNX sources building

So, which your suggestions here, taking into account points above?

FYI @xhochy  @carterbox

**Comment by xhochy** [comment]
> This ONNX cpp lib will be available for Linux only

It's unix only, not Linux only.

> The ONNX in conda-forge using dynamic protobuf - but we use static one, it can potentially lead to issues on the user side

It shouldn't, we have the same setup with onnxruntime. You can also switch to static protobuf in the ONNX feedstock. ONNX is one of the rare cases where I see an argument for linking protobuf statically (to have different global registries).

File: recipe/build.sh [hunk 4]
```diff
@@ -15,9 +15,7 @@ cmake ${CMAKE_ARGS}                                                          \
     -DCMAKE_BUILD_TYPE=Release                                               \
     -DOPENVINO_EXTRA_MODULES="$SRC_DIR/openvino_contrib/modules/arm_plugin"  \
```
Comment (ilya-lavrenov) [inline recipe/build.sh:16]: let's remove extra modules

**Comment by artanokhov** [comment]
> > This ONNX cpp lib will be available for Linux only
> 
> It's unix only, not Linux only.

Right.
But what about Windows? 

Folks, I'm just trying to understand - do we have strong concerns about this and it's declared somewhere or not?

**Comment by artanokhov** [comment]
@xhochy @carterbox
We consulted within the team and decided to proceed as follows, based on the fact that the release of OpenVINO 2023.0.0 is expected in the near future and there are external customers who are waiting for ONNH FE on Windows OS:
* Enable back OpenVINO ONNX Frontend based on sources build for upcoming 2023.0.0 release
* We planned contribution to ONNX feedstock, aligned with our further 2023.1.0 release, with following features:
  * Switch ONNX to static linkage with protobuf
  * Enable libonnx build for Windows

**Comment by carterbox** [comment]
> Enable back OpenVINO ONNX Frontend based on sources build for upcoming 2023.0.0 release
> We planned contribution to ONNX feedstock, aligned with our further 2023.1.0 release, with following features:
Switch ONNX to static linkage with protobuf
Enable libonnx build for Windows

Sounds good to me.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -77,6 +71,7 @@ requirements:
     - numpy                                  # [build_platform != target_platform]
   host:
     - tbb-devel =2021.5
+    - snappy =1.1.10
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:77]: I suppose we don't have a limitations on `snappy` version

**Comment by artanokhov** [comment]
@conda-forge-admin please rerender

**Comment by artanokhov** [comment]
@conda-forge-admin please rerender

**Comment by artanokhov** [comment]
@conda-forge-admin please rerender

**Comment by artanokhov** [comment]
@conda-forge-admin please rerender

**Comment by artanokhov** [comment]
CI affected by conda-forge/conda-forge.github.io/issues/1960

**Comment by artanokhov** [comment]
@conda-forge-admin please rerender

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**